packages/core/plugin/src/withSentryAndroid.ts

```diff
@@ -49,3 +56,58 @@ export function modifyAppBuildGradle(buildGradle: string): string {
 
   return buildGradle.replace(pattern, match => `${applyFrom}\n\n${match}`);
 }
+
+export function modifyMainApplication(config: ExpoConfig): ExpoConfig {
+  return withMainApplication(config, async config => {
+    if (!config.modResults || !config.modResults.path) {
+      warnOnce("Can't add 'RNSentrySDK.init' to Android MainApplication, because the file was not found.");
+      return config;
+    }
+
+    const fileName = path.basename(config.modResults.path);
+
+    if (config.modResults.contents.includes('RNSentrySDK.init')) {
+      warnOnce(`Your '${fileName}' already contains 'RNSentrySDK.init', the native code won't be updated.`);
+      return config;
+    }
+
+    if (config.modResults.language === 'java') {
+      if (!config.modResults.contents.includes('import io.sentry.react.RNSentrySDK;')) {
+        // Insert import statement after package declaration
+        config.modResults.contents = config.modResults.contents.replace(
+          /(package .*;\n\n?)/,
+          `$1import io.sentry.react.RNSentrySDK;\n`,
+        );
+      }
+      // Add RNSentrySDK.init
+      const originalContents = config.modResults.contents;
+      config.modResults.contents = config.modResults.contents.replace(
+        /(super\.onCreate\(\)[;\n]*)([ \t]*)/,
+        `$1\n$2RNSentrySDK.init(this);\n$2`,
+      );
+      if (config.modResults.contents === originalContents) {
+        warnOnce(`Failed to insert 'RNSentrySDK.init'.`);
+      }
+    } else {
+      // Kotlin
+      if (!config.modResults.contents.includes('import io.sentry.react.RNSentrySDK')) {
+        // Insert import statement after package declaration
+        config.modResults.contents = config.modResults.contents.replace(
+          /(package .*\n\n?)/,
+          `$1import io.sentry.react.RNSentrySDK\n`,
+        );
+      }
+      // Add RNSentrySDK.init
+      const originalContents = config.modResults.contents;
+      config.modResults.contents = config.modResults.contents.replace(
+        /(super\.onCreate\(\)[;\n]*)([ \t]*)/,
+        `$1\n$2RNSentrySDK.init(this)\n$2`,
+      );
+      if (config.modResults.contents === originalContents) {
+        warnOnce(`Failed to insert 'RNSentrySDK.init'.`);
```

Comment:
Let's add the file name to the log.

Reply:
Updated with 235f3efbe6180b41d45a748b04fd4d95393873b7